deploy/awstools/awstools.py

```diff
@@ -153,13 +158,53 @@ def construct_instance_market_options(instancemarket, spotinterruptionbehavior,
     else:
         assert False, "INVALID INSTANCE MARKET TYPE."
 
-def launch_instances(instancetype, count, instancemarket, spotinterruptionbehavior, spotmaxprice, blockdevices=None, tags=None, randomsubnet=False, user_data_file=None):
-    """ Launch count instances of type instancetype, optionally with additional
-    block devices mappings and instance tags
-
-         This will launch instances in avail zone 0, then once capacity runs out, zone 1, then zone 2, etc.
+def launch_instances(instancetype, count, instancemarket, spotinterruptionbehavior, spotmaxprice, blockdevices=None,
+                     tags=None, randomsubnet=False, user_data_file=None, timeout=timedelta(), additive=True):
+    """ Launch `count` instances of type `instancetype`
+
+    Using `instancemarket`, `spotinerruptionbehavior` and `spotmaxprice` to define instance market conditions
+    (see also: construct_market_conditions)
+
+    This will launch instances in avail zone 0, then once capacity runs out, zone 1, then zone 2, etc.
+    The ordering of availablility zones can be randomized by passing`randomsubnet=True`
+
+    Parameters
+    ----------
+    instancetype : str
+        String acceptable by `boto3.ec2.create_instances()` `InstanceType` parameter
+    count : int
+        The number of instances to launch
+    instancemarket
+    spotinterruptionbehavior
+    spotmaxprice
+    blockdevices
+    tags : dict of tag names to string values, default=None
+        Dict of tags
+    randomsubnet : bool, default=False
+        If true, subnets will be chosen randomly instead of starting from 0 and proceeding incrementally.
+    user_data_file : str, default=None
+        Path to readable file.  Contents of file are passed as `UserData` to AWS
+    timeout : datetime.timedelta, default=timedelta() (immediate timeout after attempting all subnets)
+        `timedelta` object representing how long we should continue to try asking for instances
+    additive : bool, default=True
```

Comment:
This is more of a nit, but perhaps a name like "expand_runfarm_limit", or something that gets at the fact that the goal is what is being added to, not the existing count, would be good.

Reply:
How about `expand_runfarm_by_count`?  When I read `expand_runfarm_limit`, my brain thinks that it should take an integer limit, not be a bool.

Reply:
That works for me

Reply:
As I'm plumbing this in, I'm liking `always_expand_runfarm`.  The name makes more sense when looking at the runfarm ini and I'm going to change it again unless it gives people heartburn.

@abejgonzalez, David told me to plumb it in via other channels so I'll be pushing that shortly.